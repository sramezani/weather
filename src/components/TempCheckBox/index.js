import React, { useState } from 'react';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';

import styles from './tcb.module.scss';

const useStyles = makeStyles((theme) => ({}));

export default function TempCheckBox(props) {
    const classes = useStyles();
    const [temp, setTemp] = useState(props.tempType || 'F');

    const _chnageTemp = (temp) => {
        setTemp(temp);
        props.tempCallback(temp);
    };

    return (
        <ButtonGroup disableElevation size="large" color="secondary">
            <Button variant={temp === 'C' ? 'contained' : 'outlined'} onClick={() => _chnageTemp('C')}>
                Celsius
            </Button>
            <Button variant={temp === 'F' ? 'contained' : 'outlined'} onClick={() => _chnageTemp('F')}>
                Fahrenheit
            </Button>
        </ButtonGroup>
    );
}
