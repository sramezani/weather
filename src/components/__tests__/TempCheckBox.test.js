import React from 'react';
import ReactDOM from 'react-dom';
import TempCheckBox from '../TempCheckBox';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';

jest.setTimeout(15000);

test('TempCheckBox', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TempCheckBox />, div);
});

test('renders TempCheckBox with Fahrenheit text', () => {
    const { getByText } = render(<TempCheckBox />);
    const linkElement = getByText(/Fahrenheit/i);
    expect(linkElement).toBeInTheDocument();
});

test('renders TempCheckBox with Fahrenheit text', () => {
    const { getByText } = render(<TempCheckBox />);
    const linkElement = getByText(/Celsius/i);
    expect(linkElement).toBeInTheDocument();
});

describe('TempCheckBox Button', () => {
    it('should be defined', () => {
        expect(TempCheckBox).toBeDefined();
    });
    it('should render correctly', () => {
        const tree = shallow(<TempCheckBox name="button test" />);
        expect(tree).toMatchSnapshot();
    });
});
