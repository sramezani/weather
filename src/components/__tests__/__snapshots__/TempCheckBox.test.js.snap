// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `ShallowWrapper {}`;

exports[`TempCheckBox Button should render correctly 1`] = `ShallowWrapper {}`;
